src/client/pages/OfferNew/Checkout/UpsellCard/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { QuoteBundleVariant, BundledQuote } from 'data/graphql'
+import { isSwedishAccident } from 'pages/OfferNew/utils'
+import { Card } from './Card'
+
+type UpsellCardProps = {
+  quoteBundleVariants: QuoteBundleVariant[]
+  selectedQuoteBundleVariant: QuoteBundleVariant
+  onAcceptDeal: (quoteIds: string[]) => void
+  textKeys: Record<string, any>
+}
+
+export const UpsellCard = ({
+  quoteBundleVariants,
+  selectedQuoteBundleVariant,
+  onAcceptDeal,
+  textKeys,
+}: UpsellCardProps) => {
+  const isSwedishAccidentAdded = selectedQuoteBundleVariant.bundle.quotes.some(
+    ({ quoteDetails }) => isSwedishAccident(quoteDetails),
+  )
+  if (isSwedishAccidentAdded) {
+    return null
+  }
+
+  // Look for a bundle variant that includes Accident Insurance.
+  // In case it founds it, store the bundle variant and the accident
+  // insurance quote so we can use them to accept UpsellCard deal and
+  // get the price of Accident Insurance.
```

Comment:
Is this the kind of "what" comment that we should try to avoid? Can we extract it to a named function instead?

Otherwise we need to remember to update this text when making changes to the function...

We should probably memoize this anyways!

Reply:
I made some updates to avoid the comment and improve readability